scripts/dataflow_copy.py

```diff
@@ -3,18 +3,76 @@
 
 import psycopg2
 import os
+import shutil
+import json
 
 from neurobooth_terra import Table, create_table
 from neurobooth_terra.fixes import OptionalSSHTunnelForwarder
 from neurobooth_terra.dataflow import copy_files
 
 from config import ssh_args, db_args
 
+
+# A note about setting the value of reserve_threshold in
+# the config json:
+# 
+# The reserve_threshold value must be in units of bytes
+# A specific TB value can be expressed as TB*1024**4
+# For example a reserve_threshold of 2TB can be expressed
+# as 2*1024**4 = 2199023255552
+# 
+# The value of reserve_threshold should be chosen such that
+# it is 3 to 4 times the amount of data generated per session,
+# and leaves about 2% of the total disk memory empty
+# 
+# A note about suitable_volumes in the config json:
+# The destination directory paths should exist.
+
+
+def get_volume_to_fill(volumes: list, threshold: int) -> str:
+    ''' Return the most filled volume that is below reserve_threshold
+    '''
+    vol_disk_usage = {}
+    for vol in volumes:
+        stats = shutil.disk_usage(vol)
+        if stats.free > threshold:
+            vol_disk_usage[vol] = stats.used
+
+    if len(vol_disk_usage) <= 1:
+        raise ValueError(f'Only {len(vol_disk_usage)} available volume left, aborting copy till more volumes are added')
+
+    return max(vol_disk_usage, key=vol_disk_usage.get)
+
+
+def check_if_copied(session: str, volumes: list) -> tuple:
```

Comment:
Can better annotate function signature for type checking:
`def check_if_copied(session: str, volumes: List[str]) -> Tuple[bool, str]`

Would apply this comment to other new functions as well.

Reply:
Done, thanks! Will incrementally do this for other functions.